Configuration Section (#32)
directus/docs: content/11.configuration/13.migrations.md

# Migrations

Directus allows adding custom migration files that run whenever the migration commands are run.

| Variable          | Description                          | Default Value  |
| ----------------- | ------------------------------------ | -------------- |
| `MIGRATIONS_PATH` | Where custom migrations are located. | `./migrations` |

The file name follows the following structure `[identifier]-[name].js`, for example `20201202A-my-custom-migration.js`.

Every file in the root of the `migrations` directory is treated as a migration. Files that don't include a `-` character are ignored. If you want to rely on shared helper functions between migrations, put them in a subfolder so they aren't loaded in by the migrations helper.

## Structure

Migrations have to export an `up` and a `down` function. These functions get a [Knex](http://knexjs.org) instance that can be used to do virtually whatever.

```js
export async function up(knex) {
	await knex.schema.createTable('test', (table) => {
		table.increments();
		table.string('rijk');
	});
}

export async function down(knex) {
	await knex.schema.dropTable('test');
}
```

::callout{type="warning" title="Backup Your Database"}

> Please make sure you know what you're doing and backup your database before adding custom migrations.

Reviewer: Something about "make sure you know what you're doing" doesn't sit quite right. I wonder if we can rephase to something like "Please proceed at your own risk" or something along the line of how these actions are destructive. Maybe I'm overthinking this?

Author: ```suggestion
Proceed at your own risk and backup your database before adding custom migrations.
```

::

## Migrations and Directus Schema

Migrations can be used to manage the contents of Directus collections (e.g. initial hydration). In order to do it, you must ensure that the schema is up to date before running your migrations.

> `directus database migrate:latest` runs the required Directus internal migrations and the migrations from `migrations` directory. In general, you need the following flow:

Reviewer: ```suggestion
`directus database migrate:latest` runs the required Directus internal migrations and the migrations from the `migrations` directory. In general, you need the following flow:
```
